README.md

```diff
@@ -584,6 +584,68 @@ For more information about inheriting configuration from a gem please check
 * Avoid the usage of class (`@@`) variables due to their "nasty" behavior in
   inheritance.
 
+* Avoid the usage of singleton (class, module) instance variables unless you
+  really need them, for example when writing a cache or repository, as this is
+  shared mutable state. This can cause thread-safety issues in multi-threaded
+  apps unless accounted for, e.g. by protecting the mutable state with a Mutex.
+
+  ~~~ ruby
+  # bad
+  class Processor
+    class << self
+      attr_reader :thing
+
+      def call(thing, action)
+        @thing = thing
+        thing.prepare_for_action
+        process(action)
+      end
+
+      private
+
+      def process(action)
+        Actions.for(action).perform_with(thing)
+      end
+    end
+  end
+
+  # good - same interface
+  class Processor
+    class << self
+      def call(thing, action)
+        thing.prepare_for_action
+        process(action, thing)
+      end
+
+      private
+
+      def process(action, thing)
+        Actions.for(action).perform_with(thing)
+      end
+    end
+  end
+
+  # good - changed interface with new instance per usage
+  class Processor
+    attr_reader :thing
+
+    def initialize(thing)
+      @thing = thing
+    end
+
+    def call(action)
+      thing.prepare_for_action
+      process(action)
+    end
+
+    private
+
+    def process(action)
+      Actions.for(action).perform_with(thing)
+    end
+  end
+  ~~~
```

Comment:
This is a pretty long example for a simple rule. Could we maybe rethink these code snippets and exemplify the rule with something a bit simpler?